src/utils/print.jl

```diff
@@ -40,3 +40,100 @@ function Base.show(io::IO, ::MIME"text/plain", data::PowerSystemTableData)
         println(io, "$(summary(df))")
     end
 end
+
+function Base.show(io::IO, ist::Component)
+    print(io, string(nameof(typeof(ist))), "(")
+    is_first = true
+    for (name, field_type) in zip(fieldnames(typeof(ist)), fieldtypes(typeof(ist)))
+        if field_type <: IS.TimeSeriesContainer ||
+           field_type <: InfrastructureSystemsInternal
+            continue
+        else
+            getter_func = getfield(PowerSystems, Symbol("get_$name"))
+            val = getter_func(ist)
+        end
+        if is_first
+            is_first = false
+        else
+            print(io, ", ")
+        end
+        print(io, val)
+    end
+    print(io, ")")
+end
+
+function Base.show(io::IO, ::MIME"text/plain", ist::Component)
+    default_units = false
+    if isnothing(ist.internal.units_info)
+        print(io, "\n")
+        @warn("SystemUnitSetting not defined, using NATURAL_UNITS for dispalying device specification.")
+        ist.internal.units_info =
```

Comment:
Can you use `set_unit_system!` and `get_units_setting` instead of dot accessors? I'm not sure why those names aren't symmetric. @jd-lara ? Those functions also use dot accessors, which implicitly require a field named `internal`. They could call `get_internal` instead.

Reply:
I think that is better way of doing this, I'll also look into updating those function